cmd/vfkit/main.go

```diff
@@ -198,3 +220,33 @@ func runVirtualMachine(vmConfig *config.VirtualMachine, vm *vf.VirtualMachine) e
 
 	return <-errCh
 }
+
+func startIgnitionProvisionerServer(ignitionReader io.Reader, ignitionSocketPath string) error {
+	mux := http.NewServeMux()
+	mux.HandleFunc("/", func(w http.ResponseWriter, _ *http.Request) {
+		_, err := io.Copy(w, ignitionReader)
```

Comment:
This could use https://pkg.go.dev/net/http#ServeContent (no need to make this change right, this can be a follow-up)